os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -115,19 +115,29 @@ int SOCPS_AONWakeReason(void)
 
 void pg_timer_int_handler(void *Data)
 {
-	pmvdbg("Timer wakeup interrupt handler!!\n");
+	pmvdbg("PM Timer interrupt handler!!\n");
+	switch (g_pm_timer.use_timer) {
+		case USE_PM_LOCK_TIMER:
+			pm_relax(PM_IDLE_DOMAIN, PM_NORMAL);
+			break;
+		case USE_PM_WAKEUP_TIMER:
+			pm_activity(PM_IDLE_DOMAIN, 9);
+			break;
+		default:
+			pmdbg("Timer callback triggered without setting timer, Unexpected Error!!!\n");
+			break;
+	}
 	// Reset the global struct
-    g_timer_wakeup.use_timer = 0;
-    g_timer_wakeup.timer_interval = 0;
+	g_pm_timer.use_timer = 0;
+	g_pm_timer.timer_interval = 0;
 	// Switch status back to normal mode after wake up from interrupt
-	pm_activity(PM_IDLE_DOMAIN, 9);
 }
 
-void ap_timer_helper(void) {
+void up_set_timer(void) {
 	// Check whether timer interrupt need to be set
-	if (g_timer_wakeup.use_timer) {
+	if (g_pm_timer.use_timer) {
```

Comment:
Similarly here, we are having the condition of 2 timer use case which will not have conflict, but what if later we have one-shot and periodical together? Then we will need another g_timer_2, then this method cannot work, we need bit masking method for more general working.

Reply:
Let's do a switch case here similar to the callback handler

Reply:
1. Keeping in mind that we do not need to consider periodic timers right now, does your opinion change?
2. I observe that currently, in all cases, the callback has to go to the same pg_int_handler() function, so a switch cases is not needed while starting the one-shout timer. The checking can just be done in the callback itself.
3. Again, the scope for future expansion, but modifications for that can be done if and when required.

Reply:
I think we can implement that now, despite of the future implementation grows, as the bit masking method can be generally fit into all kind of considerations.
I suggest as so is because this is a structure implementation, not a request-dependent implementation.

Reply:
Noted for future scope, in separate commits.